src/cc65/datatype.c

```diff
@@ -1037,18 +1037,113 @@ Type* ArrayToPtr (Type* T)
 
 
 
+int IsClassObject (const Type* T)
+/* Return true if this is a fully described object type */
+{
+    return !IsTypeFunc (T) && !IsClassIncomplete (T);
+}
+
+
+
+int IsClassIncomplete (const Type* T)
```

Comment:
It makes more sense to me to swap the function names `IsClassIncomplete()` and `HasUnknownSize()`.  I can't think of `void` as being "incomplete".

Also, this function is a superset of the other function.  This one simply can return the value of the other one or `IsTypeVoid()`.

Reply:
The standard requires `void` to be incomplete (https://port70.net/~nsz/c/c99/n1256.html#6.2.5p19), and `IsClassIncomplete()` is to reflect that concept. The reason why there is `HasUnknownSize()` is that  cc65 supports "void variables" as an extension. It looks more logical to me that `HasUnknownSize()` should be implemented with `IsClassIncomplete()` as `IsClassIncomplete() and !IsTypeVoid()`. What's your opinion?

Reply:
It's inefficient.  That implementation would call `IsTypeVoid()` twice.  My idea is an implementation detail rather than a semantic detail.

Reply:
What's your opinion on the naming?

As to compiler performance, my old rough profiling indicated that the heaviest CPU hog in non-optimized build was... `CHECK ()`. Surprised?

Reply:
> What's your opinion on the naming? Do you still insist on swapping the two?

No, after reading the "Types" section of C99, I agree with your names.
> 
> As to compiler performance, my old rough profiling indicated that the heaviest CPU hog in non-optimized build was... `CHECK ()`. Surprised?

Yep.